### DISCUSSION TITLE
`limit` with `ids` ergo: how to change top N tasks from a report

### DISCUSSION THREAD

**Discussion by allgreed** [GothenburgBitFactory/taskwarrior#3593] [Q&A] [answered]
- I want to `mod X` on `next limit:20 -cron` (so: `task next limit:20 -cron mod X`)
    - where `X = reviewed:now`, but I don't think it matters much, I want to mark top 20 tasks as reviewed, because I don't need to review them, since I'll have a look at them soon anyway.
 
That does not work as expected, therefore I thought I should go the id route - `task $(task [somethign] ids) mod X`. So:  `task next limit:20 -cron ids`, however this does not limit the ids returend [neither does putting the `limit` at the end].

What I currently do is `task next limit:20 - cron`, copy the text output into vim, do some processing to get a comma separated list of ids and craft the final command.

Is there a sensible way of approaching this, where I can just run 1 command and get this done? Or should look more into automating the processing of the output?

**Comment by felixschurk** [comment, marked as answer]
Hei, I did not found a way of making it work with the `ids` command. 
However what you could do, but also more on the postprocessing site is having a custom report that only outputs the ids you want with the specific limit and then pipe that into a modification.

If you add something like:
```sh
report.treport.columns           = id
report.treport.filter            = status:pending -WAITING limit:4
report.treport.sort            = urgency-
report.treport.context	          = 0
```

Now if you run your report and turn of extra verbosity with `rc.verbose:0` you should get something like that:
```sh

ID
--
18
19
20
21

```

Now cutting out the first header lines you could add a `tail -n +4` command to the pipe. So calling
```sh
$ task treport rc.verbose:0 | tail -n +4
18
19
20
21

```
Now you could call this command and modify as you like them over `task $(task treport rc.verbose:0 | tail -n +4) mod XXX`.

This way you would not need to postprocess something in vim. :)
